### DISCUSSION TITLE
Quais estratégias vocês utilizam para refatorar códigos legados?

### DISCUSSION THREAD

**Discussion by woliveiras** [frontendbr/forum#1009] [Discussão]
Existem diversas técnicas conhecidas e catalogadas para refatoração, como os livros Refactoring e Refactoring to Patterns, mas muitas vezes não utilizamos todas essas boas práticas e fazemos algo a nosso modo (por algum motivo).

Eu gostaria de saber de vocês: quais as estratégias utilizam para matar um **código legado** que é **muito utilizado em toda a aplicação**?

![Ameaçando de morte aquele código ruim](https://media.giphy.com/media/1ludrxHRnUmT6/giphy.gif)

**Comment by Cleysonlb** [comment]
Refatoração não é algo que acontece da noite pro dia, aqui onde trabalho, sempre que temos que dar manutenção em algo legado e tem possibilidade (tempo) para refatorar a gente refatora, aos poucos o código legado ta morrendo , mas temos que ter em mente que refatorar é algo eterno 😂  haha

**Comment by diessica** [comment]
Dá uma olhada: https://refactoring.guru/ meu resource favorito!

**Comment by felipefialho** [comment]
Sensacional @diessica! To precisando muito ler sobre isso, caiu como uma luva.
